.github/workflows/publish.yml

```diff
@@ -0,0 +1,57 @@
+name: publish distributions
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  # Run weekly at 1:23 UTC
+  schedule:
+    - cron: '23 1 * * 0'
+  workflow_dispatch:
+  release:
+    types:
+      - published
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  dist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: hynek/build-and-inspect-python-package@v2
+
+  publish:
+    needs: [dist]
+    environment: pypi
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'thaler-lab' && github.event_name == 'release' && github.event.action == 'published'
```

Comment:
Conditioning on `github.repository_owner == 'thaler-lab'` also requires that the release happen in the main project.